tests/intersection/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -87,13 +87,19 @@ type DuplexStreamOptions =
   };
 
 function hasObjectMode_bad(options: DuplexStreamOptions): boolean {
-  return options.objectMode || options.readableObjectMode ||
-    options.writableObjectMode; // error, undefined ~> boolean
+  return (
+    options.objectMode ||
+    options.readableObjectMode ||
+    options.writableObjectMode
+  ); // error, undefined ~> boolean
 }
 
 function hasObjectMode_ok(options: DuplexStreamOptions): boolean {
-  return !!(options.objectMode || options.readableObjectMode ||
-    options.writableObjectMode);
+  return !!((
```

Comment:
See `needsParens` in `fast-path.js`. That's where all the logic is for detecting if parens needs to be inserting or not for expressions.